sdb/command.py

```diff
@@ -240,6 +240,29 @@ def _call(self,
         """
         raise NotImplementedError()
 
+    def __input_type_check(
+            self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
```

Comment:
I think that this also needs to accept any walkable type, at least for Locator's.  see `Locator.caller()`.   How much benefit do Locator's get from this new checking?  If not much, then I think only non-Locator, non-PrettyPrinter commands would be getting new checking (of which there are not very many).  Note that Walker also has this checking built in, but that could be removed in favor of the Command type-checking.